addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,16 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            if not self.env.user._is_public():
+                messages_per_partner[self.env.user.partner_id] |= message
+            if message.author_id:
```

Comment:
I wonder if we shouldn't also make the `_is_public` check for author, as it could also be a public user (eg. for messages from livechat).

Other than that, it looks good now, thanks for the work!

Already setting task green for testing.

Reply:
Done!!